xcdat/spatial_avg.py

```diff
@@ -441,40 +500,38 @@ def _align_longitude_to_360_axis(self, domain_bounds: xr.DataArray):
         This method returns ``domain_bounds`` that are intended for calculating
         spatial weights only.
         """
-
-        # copy bounds to work on
         d_bounds = domain_bounds.copy()
-        # find instances of d_bounds spanning the prime meridian
-        pmb_index = np.where(d_bounds[:, 1] - d_bounds[:, 0] < 0)[0]
-        # check bounds to ensure they range from 0 to 360
+        # TODO: Refactor this check
         if (np.min(d_bounds).values < 0) | (np.max(d_bounds).values > 360):
             raise ValueError(
                 "Longitude bounds between 0 and 360. Use _swap_lon_axes"
                 "before calling _align_longitude_to_360_axis."
             )
 
+        # FIXME: Apply default value as None
+        prime_meridian = np.where(d_bounds[:, 1] - d_bounds[:, 0] < 0)[0]
         # there should be 0 or 1 such grid cells
-        if len(pmb_index) > 1:
+        if len(prime_meridian) > 1:
             raise ValueError("More than one grid cell spans prime meridian.")
-        elif len(pmb_index) == 1:
+        elif len(prime_meridian) == 1:
             # convert index array to integer index
-            pmb_index = pmb_index[0]
+            prime_meridian = prime_meridian[0]
             # reorient bound to span across zero (i.e., [361, 1] -> [-1, 1])
-            d_bounds[pmb_index, 0] = d_bounds[pmb_index, 0] - 360.0
+            d_bounds[prime_meridian, 0] = d_bounds[prime_meridian, 0] - 360.0
             # extend the dataarray to nlon+1 by concatenating the grid cell
             # that spans the prime meridian to the end
-            d_bounds = xr.concat((d_bounds, d_bounds[pmb_index, :]), dim="lon")
+            d_bounds = xr.concat((d_bounds, d_bounds[prime_meridian, :]), dim="lon")
             # produce an equivalent bound that spans 360
             # (i.e., [-1, 1] -> [359, 361])
-            repeat_bound = d_bounds[pmb_index, :] + 360.0
+            repeat_bound = d_bounds[prime_meridian, :] + 360.0
             # place this repeat bound at the end of the DataArray
             d_bounds[-1, :] = repeat_bound
             # limit bounds to [0, 360]
-            d_bounds[pmb_index, 0] = 0.0
+            d_bounds[prime_meridian, 0] = 0.0
             d_bounds[-1, 1] = 360.0
         else:
-            pmb_index = None
-        return pmb_index, d_bounds
+            prime_meridian = None
+        return d_bounds, prime_meridian
```

Comment:
The return values are swapped since this method operates on the `domain_bounds` if a `prime_meridian` exists.